File: docs/breez-sdk/snippets/go/buying_bitcoin.go
package example

import (
	"log"

	"github.com/breez/breez-sdk-spark-go/breez_sdk_spark"
)

func BuyBitcoin(sdk *breez_sdk_spark.BreezSdk) error {
	// ANCHOR: buy-bitcoin
	optionalLockedAmountSat := uint64(100_000)
	optionalRedirectUrl := "https://example.com/purchase-complete"

	request := breez_sdk_spark.BuyBitcoinRequestMoonpay{
		LockedAmountSat: &optionalLockedAmountSat,
		RedirectUrl:     &optionalRedirectUrl,
	}

	response, err := sdk.BuyBitcoin(request)
	if err != nil {
		return err
	}

	log.Printf("Open this URL in a browser to complete the purchase:")
	log.Printf("%v", response.Url)
	// ANCHOR_END: buy-bitcoin
	return nil
}

func BuyBitcoinViaCashapp(sdk *breez_sdk_spark.BreezSdk) error {
	// ANCHOR: buy-bitcoin-cashapp
	// Cash App requires the amount to be specified up front.
	amountSats := uint64(50_000)

	request := breez_sdk_spark.BuyBitcoinRequestCashApp{
		AmountSats: amountSats,
	}

	response, err := sdk.BuyBitcoin(request)
	if err != nil {
		return err
	}

	log.Printf("Open this URL in Cash App to complete the purchase:")
	log.Printf("%v", response.Url)
	// ANCHOR_END: buy-bitcoin-cashapp
	return nil
}

File: docs/breez-sdk/snippets/python/src/buying_bitcoin.py
# pylint: disable=duplicate-code
import logging
from breez_sdk_spark import (
    BreezClient,
    BuyBitcoinRequest,
)


async def buy_bitcoin(client: BreezClient):
    # ANCHOR: buy-bitcoin
    # Optionally, lock the purchase to a specific amount
    optional_locked_amount_sat = 100_000
    # Optionally, set a redirect URL for after the purchase is completed
    optional_redirect_url = "https://example.com/purchase-complete"

    try:
        request = BuyBitcoinRequest(
            locked_amount_sat=optional_locked_amount_sat,
            redirect_url=optional_redirect_url,
        )

        response = await client.buy_bitcoin(request=request)
        logging.debug("Open this URL in a browser to complete the purchase:")
        logging.debug(response.url)
    except Exception as error:
        logging.error(error)
        raise
    # ANCHOR_END: buy-bitcoin
